File: 0055-jump-game/0055-jump-game.rb
# @param {Integer[]} nums
# @return {Boolean}
def can_jump(nums)
  max_reach = i = 0
  while i <= max_reach && max_reach < nums.size
      max_reach = [max_reach, i + nums[i]].max
      i += 1
  end
  return max_reach >= nums.size - 1
end

File: 0055-jump-game/0055-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int maxr = 0;
        int i = 0;
        while (i<=maxr) {
            maxr = max(maxr, i + nums[i]);
            i++;
            if (maxr >= nums.size()- 1) return true;
        }
        return false;
    }   
};
